### ISSUE TITLE
unexpected results in Top Transactions for Time Period [APM] visualization

### ISSUE THREAD

**Issue by LeeDr** [elastic/apm-server#1732]
**Version:** 6.x and master (7.0)

I've been testing apm in the integration-test repo since the 5.6 release.  The test has always done these steps;

1. download https://raw.githubusercontent.com/elastic/apm-server/master/testdata/transaction/payload.json
1. load that data into apm-server with `curl -H "Content-type: application/json" -d @payload.json http://localhost:8200/v1/transactions -v`
1. In Kibana navigate to visualization `Top Transactions for Time Period [APM]`
1. check expected result `'GET /api/types\nrequest 19 33 View Spans'`

But on 6.x I now get;
`'GET /api/types\nrequest 18,633 32,592 View Spans'`

@graphaelli suggested I switch on master branch to downloading;
https://raw.githubusercontent.com/elastic/apm-server/1b0437b2b740e21e2017b397b0cd307d0921521e/testdata/intake-v2/transactions_spans.ndjson

and using the v2 api like;
`curl -H "Content-type: application/x-ndjson" --data-binary @transactions_spans.ndjson http://localhost:8200/intake/v2/events -v`

But I still get the same results which don't match previous results.

I'm not sure how to troubleshoot this any further.

**Comment by graphaelli** [comment]
I also said it wouldn't fix the test, just that the v1 api is about to be removed so this test will break horribly real soon now, as the data in payload hasn't changed.

The `19 -> 18,633, 33 -> 32,592` is due to changes in the format of the duration fields from #1717.  I'm not sure why the top transaction is apparently returning a different hit now though.

**Comment by simitt** [comment]
I agree with @graphaelli that the changed values are expected with PR #1717 getting merged, for `master` and `6.x`. 

The issue that no spans are found for the chosen transaction, I can only reproduce on `master`. Querying in `6.x` still returns the 4 related `spans`. 
Will look into what changed here in master.

**Comment by simitt** [comment]
It looks like this line ` \"params\": {\"customLabel\": \"View Spans\", \"sortOrder\": \"desc\", \"size\": 1, \"field\": \"transaction.id\", \"aggregate\": \"concat\", \"sortField\": \"@timestamp\"}}]` from https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/server/tutorials/apm/saved_objects/saved_objects.json#L165 is not working properly in master. 
This also happened before #1717, so this is unrelated to the duration format change. 

@LeeDr could you please link to the testcase and the failures on Jenkins, so we can narrow down the changes that happened. To me this looks like something in Kibana or ES might have changed, so the sorting is not respected any more.

**Comment by simitt** [comment]
The change to `microseconds` works as expected . The issue that the sort order is not properly working in one of the dashboards can be ignored for now, as we are removing bundling the dashboards and visualisations from Kibana and the APM Server as pointed out in #1673. 

@LeeDr are you therefore ok with closing this issue?
